gensim/models/word2vec_corpusfile.pyx

```diff
@@ -225,7 +226,10 @@ cdef void prepare_c_structures_for_batch(
 
     # precompute "reduced window" offsets in a single randint() call
     for i in range(effective_words[0]):
-        reduced_windows[i] = random_int32(next_random) % window
+        if do_reduced_windows:
+            reduced_windows[i] = random_int32(next_random) % window
```

Comment:
This doesn't look like `[1, window]`. Or rather, if `reduced_windows` is expected to contain "window minus one", shouldn't your new code path be `reduced_windows[i] = window - 1`?

Reply:
The *effective* window winds up being from `1` to `window` (inclusive), because these `0` to `window - 1` values get offset against a full `window` sum, per the indexing math just a few lines down: https://github.com/RaRe-Technologies/gensim/blob/6afb3ff9bcaa4f5faef05a4223c324852b2e4c22/gensim/models/word2vec_inner.pyx#L589-L594

(I suspect it's this way because it was mimicking something in the original C code - it could be clearer if `reduced_windows` values became the literal, actually-used window size, by adding 1, changing the later indexing to match.)

Reply:
That's what I thought – hence my off-by-one warning for the new code path.

Reply:
For the variable-size window, the rng code is the same as in the current `develop` branch.
But you are right, there is a mistake here: in the case when we want fixed-size windows, it should be `reduced_windows[i] = 0`.
This change will be part of an upcoming commit.